compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -4577,6 +4582,45 @@ private BLangExpression createConditionForErrorFieldMatchPatterns(
         return statementExpression;
     }
 
+    private BType createMatchingRecordType(BLangErrorFieldMatchPatterns errorFieldMatchPatterns) {
+        BRecordTypeSymbol detailRecordTypeSymbol = new BRecordTypeSymbol(
+                SymTag.RECORD,
+                Flags.PUBLIC,
+                names.fromString("$anonErrorType$" + UNDERSCORE + recordCount++ + "$detailType"),
```

Comment:
It is better if we can move this `$anon` name gen to `BLangAnonymousModelHelper`.